infrastructure/src/main/java/org/corfudb/infrastructure/management/FailureDiscoveryPolicy.java

```diff
@@ -0,0 +1,35 @@
+package org.corfudb.infrastructure.management;
+
+/**
+ * {@link FailureDiscoveryPolicy} is the enumeration of policies each of which representing
+ * concrete algorithms for suspecting the failure or healing of specific servers in a Corfu
+ * cluster of servers.
+ *
+ * Created by Sam Behnam on 10/19/18.
+ */
+public enum FailureDiscoveryPolicy {
+    /**
+     * FULLY_CONNECTED_CLUSTER represents a failure and healing detection approach in which the
```

Comment:
Not sure if i understood the name of the class ? This seems to be providing a healing policy as well ?

Reply:
Yes, failure and healing are the complementary algorithms both form a policy together. The healing is implicit as discovering lack of failure is implied as healing. I can change to other names if you have suggestion for it.